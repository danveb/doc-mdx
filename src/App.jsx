import { useState, createContext } from "react";
import { About, Navbar, Post } from "./components";

// createContext
export const ThemeContext = createContext();

export default function App() {
  // useState 
  const [darkMode, setDarkMode] = useState(true);

  // value to be darkMode, setDarkMode
  const value = {
    darkMode, setDarkMode
  }

  return (
    <ThemeContext.Provider value={value}>
      <Navbar />
      <About />
      <Post />
    </ThemeContext.Provider>
  )
}
