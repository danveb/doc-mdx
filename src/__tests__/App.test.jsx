import { describe, test, expect } from "vitest";
import { render } from "@testing-library/react";
import App from "../App";
import { BrowserRouter } from "react-router-dom";

describe("App component", () => {
  test("renders without crashing", () => {
    const { container } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );
    expect(container).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});