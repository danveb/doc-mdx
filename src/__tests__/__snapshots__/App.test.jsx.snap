// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="navbar"
  >
    <div
      class="nav__wrapper container"
    >
      <div
        class="nav__logo"
      >
        <svg
          class="w-6 h-6"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14.25 9.75 16.5 12l-2.25 2.25m-4.5 0L7.5 12l2.25-2.25M6 20.25h12A2.25 2.25 0 0 0 20.25 18V6A2.25 2.25 0 0 0 18 3.75H6A2.25 2.25 0 0 0 3.75 6v12A2.25 2.25 0 0 0 6 20.25Z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <p>
          doc-mdx
        </p>
        <button
          class="nav__icon"
        >
          <svg
            class="w-6 h-6"
            fill="#fff"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 2.25a.75.75 0 0 1 .75.75v2.25a.75.75 0 0 1-1.5 0V3a.75.75 0 0 1 .75-.75ZM7.5 12a4.5 4.5 0 1 1 9 0 4.5 4.5 0 0 1-9 0ZM18.894 6.166a.75.75 0 0 0-1.06-1.06l-1.591 1.59a.75.75 0 1 0 1.06 1.061l1.591-1.59ZM21.75 12a.75.75 0 0 1-.75.75h-2.25a.75.75 0 0 1 0-1.5H21a.75.75 0 0 1 .75.75ZM17.834 18.894a.75.75 0 0 0 1.06-1.06l-1.59-1.591a.75.75 0 1 0-1.061 1.06l1.59 1.591ZM12 18a.75.75 0 0 1 .75.75V21a.75.75 0 0 1-1.5 0v-2.25A.75.75 0 0 1 12 18ZM7.758 17.303a.75.75 0 0 0-1.061-1.06l-1.591 1.59a.75.75 0 0 0 1.06 1.061l1.591-1.59ZM6 12a.75.75 0 0 1-.75.75H3a.75.75 0 0 1 0-1.5h2.25A.75.75 0 0 1 6 12ZM6.697 7.757a.75.75 0 0 0 1.06-1.06l-1.59-1.591a.75.75 0 0 0-1.061 1.06l1.59 1.591Z"
            />
          </svg>
        </button>
      </div>
    </div>
  </header>
  <div
    class="about"
  >
    <div
      class="about__wrapper text__container"
    >
      <div>
        <div
          class="about__main"
        >
          <p
            class="title"
          >
            Danny Bae
          </p>
          <p
            class="subtitle"
          >
            Frontend Developer
          </p>
        </div>
        <div
          class="about__sub"
        />
      </div>
      <div>
        <div
          class="about__main"
        />
        <div
          class="about__sub"
        >
          <p
            class="title"
          >
            Hello World
          </p>
          <p
            class="text"
          >
            This is doc-mdx, a simple application that allows users create concise technical documentation directly from a markdown document. In this article, I will go over the process of installation and usage.
          </p>
        </div>
      </div>
      <div>
        <div
          class="about__main"
        />
        <div
          class="about__sub"
        >
          <p
            class="title"
          >
            Markdown
          </p>
          <p
            class="text"
          >
            Markdown is a lightweight markup language widely used for formatting plain text documents. It is used among developers who need a simple and highly readable way to craft documentation related to a code base. Users can use markdown syntax or even HTML syntax to build documents.
          </p>
        </div>
      </div>
      <div>
        <div
          class="about__main"
        />
        <div
          class="about__sub"
        >
          <p
            class="title"
          >
            doc-mdx
          </p>
          <p
            class="text"
          >
            This project incorporates markdown into React.js.
          </p>
        </div>
      </div>
    </div>
  </div>
  <article
    class="article"
  >
    <div
      class="article__wrapper container"
    >
      <div
        class="post__wrapper"
      >
        <span />
      </div>
    </div>
  </article>
  <div
    class="footer"
  >
    <div
      class="footer__wrapper container"
    >
      <p
        class="mini__text"
      >
        Â© All rights reserved |  danveb
      </p>
      <a
        href="https://www.github.com/danveb"
        rel="noreferrer noopener"
        target="_blank"
      >
        <img
          alt="github logo"
          src="/src/components/assets/github-mark-white.png"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;
